src/mlpack/tests/gan_test.cpp

```diff
@@ -215,7 +216,10 @@ BOOST_AUTO_TEST_CASE(GANMNISTTest)
       discriminatorPreTrain, multiplier);
 
   Log::Info << "Training..." << std::endl;
-  gan.Train(optimizer);
+  double objVal = gan.Train(optimizer);
+
+  // Test that objective value returned by GAN::Train() is finite.
+  BOOST_REQUIRE_EQUAL(boost::math::fpclassify(objVal), FP_NAN);
```

Comment:
Hmm, if this gave a loss of NaN, try maybe reducing `numIterations` to something like 100 or 1000, and maybe making the step size smaller.

Reply:
Okay, I have digged in somewhat and found following things.
1. We are only training on 10 training samples.
2. Number of iterations(`numIterations`) is only two.

Also I found that the layers of `generator` and `discriminator` are  same in `DCGAN`, `WGAN` and `WGANP` while  they are different in `StandardGan` testcase. 
Whereas , other parameters like `multiplier` , `iterations` etc. are same in all four. So I copied the
`generator` and `discriminator` layers of `DCGAN` in `standardGAN` testcase. And surprisingly, the error got eliminated. 
So there can be any of the following,
a) `parameters` of `StandardGan` are wrong.
b) `Layers` of `StandardGan`  are wrong.

However, I also read comments on top of testcase. It read `oreilly test on gan implementation` in `StandardGan` , whereas there were no such comments in other three. Hence I am not sure that the changes I have made is  a prefect solution.
If that's wrong solution just let me know. If I am going wrong it will be great help if you provide good resource link `oreilly test`, so that I can go more deeper.

Reply:
Looks like you got it working, thanks for digging in.  It seems like nothing else is actually tested in that test case, which is a bit strange and leads me to wonder if it could be removed, but we can leave it there for now I guess.

Reply:
@rcurtin That test is primarily an indicator of the working of the `GAN` as a whole. The one above it checks for the logical correctness. This is a generative network, so if you want to check the results, you can just save the output from this test, and see for yourself. Ideally, this should have been moved to the models repository, but if we had actually done that, we wouldn't have caught the NaN error if we simply merged this PR right? That would have lead to further confusion.